pkg/apis/sources/v1alpha1/sinkbinding_lifecycle.go

```diff
@@ -84,37 +85,68 @@ func (sb *SinkBinding) Do(ctx context.Context, ps *duckv1.WithPod) {
 		return
 	}
 
+	var ceOverrides string
+	if sb.Spec.CloudEventOverrides != nil {
+		if co, err := json.Marshal(sb.Spec.SourceSpec.CloudEventOverrides); err != nil {
+			logging.FromContext(ctx).Error(fmt.Sprintf("Failed to marshal CloudEventOverrides into JSON for %+v, %v", sb, err))
+		} else if len(co) > 0 {
+			ceOverrides = string(co)
+		}
+	}
+
 	spec := ps.Spec.Template.Spec
 	for i := range spec.InitContainers {
 		spec.InitContainers[i].Env = append(spec.InitContainers[i].Env, corev1.EnvVar{
 			Name:  "K_SINK",
 			Value: uri.String(),
 		})
+		spec.InitContainers[i].Env = append(spec.InitContainers[i].Env, corev1.EnvVar{
+			Name:  "K_CE_OVERRIDES",
+			Value: ceOverrides,
+		})
 	}
 	for i := range spec.Containers {
 		spec.Containers[i].Env = append(spec.Containers[i].Env, corev1.EnvVar{
 			Name:  "K_SINK",
 			Value: uri.String(),
 		})
+		spec.Containers[i].Env = append(spec.Containers[i].Env, corev1.EnvVar{
+			Name:  "K_CE_OVERRIDES",
+			Value: ceOverrides,
+		})
 	}
 }
 
 func (sb *SinkBinding) Undo(ctx context.Context, ps *duckv1.WithPod) {
 	spec := ps.Spec.Template.Spec
 	for i, c := range spec.InitContainers {
+		if len(c.Env) == 0 {
+			continue
+		}
+		env := make([]corev1.EnvVar, 0)
```

Comment:
If you want to be pedantic, you could `-2`, but then we need to `max(0, that)` and it gets unwieldy.
```suggestion
		env := make([]corev1.EnvVar, len(spec.InitContainers[i].Env))
```

Reply:
It does not work:

```
--- FAIL: TestSinkBindingDoNoURI (0.00s)
    sinkbinding_lifecycle_test.go:492: Undo (-want, +got):   &v1.WithPod{
          	TypeMeta:   v1.TypeMeta{},
          	ObjectMeta: v1.ObjectMeta{},
          	Spec: v1.WithPodSpec{
          		Template: v1.PodSpecable{
          			ObjectMeta: v1.ObjectMeta{},
          			Spec: v1.PodSpec{
          				Volumes:        nil,
          				InitContainers: nil,
          				Containers: []v1.Container{
          					{
          						... // 5 identical fields
          						Ports:   nil,
          						EnvFrom: nil,
          						Env: []v1.EnvVar{
        + 							{},
        + 							{},
          						},
          						Resources:    v1.ResourceRequirements{},
          						VolumeMounts: nil,
          						... // 12 identical fields
          					},
          				},
```

Reply:
ah it needed `...,0, len(x))